Author: Niccolo Rigacci <niccolo@rigacci.org>
Description: Fix wifi handling with kernel 2.6.34+
 This patch fixes wifi power management if kernel 2.6.34 or newer kernels are
 used.
Debian-Bug: http://bugs.debian.org/647003
Last-Update: 2011-10-30
--- a/src/plugins/openmoko_powercontrol/plugin.vala
+++ b/src/plugins/openmoko_powercontrol/plugin.vala
@@ -181,6 +181,10 @@
     if ( !ignore_wifi )
     {
         var wifi = Path.build_filename( drivers, "s3c2440-sdi" );
+          if ( !FsoFramework.FileHandling.isPresent( wifi ) )
+          {
+        	wifi = Path.build_filename( drivers, "s3c-sdi" );
+          }
         if ( FsoFramework.FileHandling.isPresent( wifi ) )
         {
             var o = new Openmoko.WiFiPowerControl( subsystem, wifi );
